---
title: 'The Palestinian Covid Conflict'
author: "Waleejah Rana"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable_PRJNA669945.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

# Background and Overview


The outbreak of the SARS-CoV-2 infection was catastrophic for the people in Palestine. Palestinians were in amidst a cloudy political climate with the people of Israel. The Israeli-Palestinian conflict not only became the reason for the increased cases per day, this conflict expanded SARS-CoV-2 geographically as well as exponentially. The objective of this analysis was to assess the number of cases and their locations in correlation with the Israeli-Plaestinan conflict. This was done by …….. My findings concluded that …
This is a report on SARS-CoV-2, including some variant analysis [@koyama2020variant].

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r}
ggplot(vcf_with_metadata, aes(x = as.numeric(qual))) + geom_histogram()
```


```{r sample-plot}
# create a plot of SNP quality average within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the grey background
```

```{r plot-transportation-types}
# create a plot of the transportation types to include in knitted
# document and save that figure to the output/figures directory as a png
transportation_type_plot <- ggplot(data = tallied_subset,
                       aes(x = geo_type,
                           y = n,
                           fill = transportation_type)) +
geom_col(position = position_dodge()) +
labs(title = paste("Number of cities and/or counties in",
                   state_to_analyze,
                   "with mobility data"),
     x = "Type of areas",
     y = "Number of areas")
ggsave(plot = transportation_type_plot,
       filename = paste0("output/figures/",
                         core_name,
                         "_",
                         state_no_spaces,
                         "_transportation_type_plot.png"))
transportation_type_plot
```


```{r plot-time-series}
# produce a line plot of relative driving mobility across the state with date
# on the x axis and mean relative mobility on the y axis
timeseries_plot <- long_data %>%
  filter(transportation_type == "driving") %>%
  group_by(date) %>%
  summarize(mean_mobility = mean(relative_mobility)) %>%
  ggplot(aes(x = lubridate::ymd(date), y = mean_mobility)) + geom_line() +
  labs(title = paste("Statewide mean relative mobility driving levels in",
                     state_to_analyze,
                     "during COVID-19"),
       x = "Date",
       y = "Mean relative mobility")
ggsave(plot = timeseries_plot,
       filename = paste0("output/figures/",
                         core_name,
                         "_",
                         state_no_spaces,
                         "_timeseries_plot.png"))
timeseries_plot
```

```{r long-data-plot}
# create a plot of the transportation types to include in knitted
# document and save that figure to the output/figures directory as a png
long_scatterplot <- ggplot(data = tallied_subset,
                           aes(color = geo_type,
                               x = transportation_type,
                               y = n)) +
  geom_point(shape = 18) +
  labs(title = paste("Numbers of areas that have 
                     types of mobility in",
                     state_to_analyze,
                     "with mobility data"),
       x = "Type of Transportation",
       y = "Number of areas")
long_scatterplot
```




**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

# Tables

```{r example-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
