---
title: 'The Palestinian Covid Conflict'
author: "Waleejah Rana"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable_PRJNA669945.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

# Background and Overview


The outbreak of the SARS-CoV-2 infection was catastrophic for the people in Palestine. Palestinians were in amidst a cloudy political climate with the people of Israel. The Israeli-Palestinian conflict not only became the reason for the increased cases per day, this conflict expanded SARS-CoV-2 geographically as well as exponentially across the nations. The objective of this analysis was to assess the differences in SARS-CoV-2 data from settlements of Israel in contrast to settlements of Palestine regarding the Israeli-Palestinian conflict. This was done by …….. My findings concluded that …
This is a report on SARS-CoV-2, including some variant analysis [@koyama2020variant]. From a historical perspective, the ethnic cleansing of Palestinians further legitimize the existence of a colonial framework. It was horrific to see and hear the depictions of Palestinians being forced to leave at gunpoint. To the Palestinians, most did not feel to be at war, and were oppressed in various forms. The portrayal of Deir yassin, from family relatives and the documentary, made me understand the significance of persecution that Paletinans faced in the era. I found it odd to how scholars debated on whether ethnic cleansing was the root of these atrocious crimes. In like manner, the images of the Trail of Tears solidified the idea of mass colonization in this region. To not acknowledge this, is to to not acknowledge the obvious, which is the evidence of the forced removal of an indegenous people form their homeland. Knowing this, is there really any more reason to doubt the existence of an aprteid state because of the logistics behind the term: settler colonialism. 

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r}
# create histogram to display average quality number in SNPs
ggplot(vcf_with_metadata,
       aes(x = as.numeric(qual))) +
  geom_histogram()
```


```{r plot}
# create a plot of average spot length within each gene
vcf_with_metadata %>%
  group_by(gene, AvgSpotLen) %>%
  tally() %>% # this gives a column n for the average spot length for each gene
  group_by(gene) %>%
  tally() %>% 
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Average Spot length Within Each Gene",
         x = "Gene Identified") +
  theme_few() # get rid of the grey background
```


```{r plot-time-series}
# produce a line plot of relative SNPs across Palestine with date
# on the x axis and mean SNPs on the y axis
  vcf_with_metadata %>%
  group_by(Collection_Date, pos) %>%
  tally() %>%
  group_by(Collection_Date) %>%
  tally() %>%
  ggplot(aes(x = lubridate::ymd(Collection_Date),
             y = n)) +
       geom_point() +
       labs(title = paste("SARS-Cov-2 SNP Changes in Palestine"),
            x = "Date",
            y = "Observed SNPs")
```


```{r N gene SNPs within Israeli Territory and Palestinian territory}
vcf_with_metadata %>%
  tidyr::separate(col = geo_loc_name,
                  into = c("country", "city"),
                  sep = ": ") %>%
    tidyr::separate(col = city,
                  into = c("city1", "city2"),
                  sep = "\\\\, ") %>%
filter(gene == "N") %>%
select(gene, pos, alt, ref, city1, city2,) %>%
group_by(gene, pos, alt, ref, city1, city2,) %>%
tally() %>%
arrange(desc(n)) %>%
head (10) %>%
arrange(pos) %>%
knitr::kable(col.names = c("Gene",
                        "Position",
                        "Reference",
                        "Variation",
                        "city1",
                        "city2",
                        "n"))
        
```



**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

# Tables

```{r table}
# table showing the position of the SNP depending on the unique release and collection date
vcf_with_metadata %>%
  arrange(vcf_with_metadata$Collection_Date) %>%
  select(Collection_Date, ReleaseDate, alt, gene, pos) %>%
  unique() %>%
  head (30) %>%
  knitr::kable(col.names = c("Collection Date",
                             "Release Date",
                             "Gene",
                             "SNP",
                             "Position"),
                              row.names = FALSE,) 

```

```{r ORF3a--gene-mutation}
# create a table to display the occurrences of the ORF3a gene
vcf_with_metadata %>%
filter(gene == "ORF3a") %>%
select(gene, pos, alt, ref) %>%
group_by(gene, pos, alt, ref) %>%
tally() %>%
arrange(desc(n)) %>%
head (10) %>%
arrange(pos) %>%
knitr::kable(col.names = c("Gene",
                        "Position",
                        "Reference",
                        "Variation",
                        "Occurrences"
                        ))
```

**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
